File: app/src/main/java/com/academy/sda/checkers/model/Field.java
package com.academy.sda.checkers.model;

public class Field {

    private int row;
    private int column;


    public Field(int row, int column) {
        this.row = row;
        this.column = column;
    }

    public int getRow() {
        return row;
    }

    public int getColumn() {
        return column;
    }

    public boolean isBlack() {
        return (row + column) % 2 == 0;
    }

    public boolean isNeighbour(Field anotherField) {
        return Math.abs(row - anotherField.getRow()) <= 1 &&
                Math.abs(column - anotherField.getColumn()) <= 1 &&
                !this.equals(anotherField);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Field that = (Field) o;

        return row == that.row && column == that.column;

    }

    @Override
    public int hashCode() {
        int result = row;
        result = 31 * result + column;
        return result;
    }

    @Override
    public String toString() {
        return "Field{" + row+"," + column + '}';
    }
}

File: app/src/main/java/com/academy/sda/checkers/model/Field.kt
package com.academy.sda.checkers.model

class Field(val row: Int, val column: Int) {

    val isBlack: Boolean
        get() = (row + column) % 2 == 0

    fun isNeighbour(anotherField: Field): Boolean {
        return Math.abs(row - anotherField.row) <= 1 &&
                Math.abs(column - anotherField.column) <= 1 &&
                this != anotherField
    }

    override fun toString(): String {
        return "Field{$row,$column}"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false

        other as Field

        if (row != other.row) return false
        if (column != other.column) return false

        return true
    }

    override fun hashCode(): Int {
        var result = row
        result = 31 * result + column
        return result
    }


}
